environment-api/src/test/java/com/sequenceiq/environment/api/v1/environment/model/request/aws/AwsDiskEncryptionParametersTest.java

```diff
@@ -0,0 +1,115 @@
+package com.sequenceiq.environment.api.v1.environment.model.request.aws;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Set;
+
+import javax.validation.ConstraintViolation;
+
+import org.hibernate.validator.HibernateValidator;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.springframework.validation.beanvalidation.LocalValidatorFactoryBean;
+
+class AWSDiskEncryptionParametersTest {
+
+    private static final String ENCRYPTION_KEY_ARN = "encryptionKeyArn";
+
+    private LocalValidatorFactoryBean localValidatorFactory;
+
+    @BeforeEach
+    void setUp() {
+        localValidatorFactory = new LocalValidatorFactoryBean();
```

Comment:
You are testing the whole Pattern validator here not only your validation regexp.
It would be more concise if you just extract your regexp and run the checks in `encryptionKeyArnValidationTest` against the regexp only (i.e. matches or not)

Reply:
I have made the changes accordingly. Hope now this looks good.

Reply:
@bergerdenes  I am late to the party but would like to disagree. Testing the validator itself is more valuable (like an integration test), that way we can be sure that annotations are not screwed up and so the validation will be indeed applied correctly. Satya's original approach is a direct adaptation of [AzureResourceEncryptionParametersTest](https://github.com/hortonworks/cloudbreak/blob/master/environment-api/src/test/java/com/sequenceiq/environment/api/v1/environment/model/request/azure/AzureResourceEncryptionParametersTest.java), for which I must be blamed. 😄 There are also some further examples where `LocalValidatorFactoryBean` was used in a similar fashion.

In short, I suggest reverting to the original code.

Reply:
Resolving this conversation as per https://github.com/hortonworks/cloudbreak/pull/11024#discussion_r668716966